src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1222,31 +1244,22 @@ private void visitProperty(String propName, ObjectType objType, boolean isStatic
      * e.g. different return types. It also pretends that functions exist that do not.
      */
     private void emitStaticOverloads(String propName, ObjectType objType, JSType propertyType) {
-      if (!objType.isFunctionType()) {
-        return;
-      }
-      ObjectType superType = getSuperType((FunctionType) objType);
-      if (superType == null || superType.getConstructor() == null) {
-        return;
-      }
-      FunctionType superTypeCtor = superType.getConstructor();
-      if (!superTypeCtor.hasProperty(propName)) {
-        return;
-      }
+      Set<JSType> incompatibleSuperTypes = new LinkedHashSet<>();
+      collectIncompatibleSuperTypes(objType, propName, incompatibleSuperTypes);
+
       // "I, for one, welcome our new static overloads." - H.G. Wells.
-      JSType superPropType = superTypeCtor.getPropertyType(propName);
-      if (!propertyType.isSubtype(superPropType)) {
-        // If this field is public, but the super field is private, and thus not emitted, it's ok.
-        if (isPrivateProperty(superTypeCtor, propName)) return;
-        if (!propertyType.isFunctionType() || !superPropType.isFunctionType()) {
-          errors.add(JSError.make(currentSymbol.getNode(), CLUTZ_OVERRIDDEN_STATIC_FIELD,
-              objType.getDisplayName(), propName));
-        } else {
-          emit("/** WARNING: emitted for non-matching super type's static method. "
-              + "Only the first overload is actually callable. */");
-          emitBreak();
+      for (JSType superPropType : incompatibleSuperTypes) {
+        if (!propertyType.isSubtype(superPropType)) {
```

Comment:
move this check into the TODO in collectIncompatibleSuperTypes. If the type is a subtype, it is *compatible* and not incompatible.